src/engine/strat_engine/backstore/metadata/bda.rs

```diff
@@ -272,10 +276,49 @@ impl BDA {
     where
         F: Read + Seek + SyncAll,
     {
-        StaticHeader::setup(f).map(|sh| sh.map(|sh| (sh.pool_uuid, sh.dev_uuid)))
+        match StaticHeader::setup(f) {
+            Ok(Some(SetupResult::Ok(sh))) => Ok(Some((sh.pool_uuid, sh.dev_uuid))),
+            Ok(Some(SetupResult::OkWithError(sh, err))) => {
+                setup_warn(&sh, err);
+                Ok(Some((sh.pool_uuid, sh.dev_uuid)))
+            }
+            Ok(None) => Ok(None),
+            Err(err) => Err(err),
+        }
     }
 }
 
+fn setup_warn(header: &StaticHeader, err: StratisError) {
+    warn!(
+        "Experienced an I/O error while attempting to repair an ill-formed, \
+         unreadable, or stale signature block: {:?}.\n\
+         Returning device \"{:?}\" from pool \"{:?}\".",
+        err, header.dev_uuid, header.pool_uuid
+    );
+}
+
+fn bda_write_check<F>(
+    f: &mut F,
+    bda_buf: &[u8],
```

Comment:
Please change ```bda_buf``` to ```sh_buf```. I know the original calls it ```bda_buf```, but it's wrong, and since this is new code we might as well get it right here.